### PR TITLE
Always convert explicitly longitudes and latitudes to 'rad' instead of 'radians'

### LINKED ISSUE #14580: AttributeError: Longitude instance has no attribute 'radian' (but only from pytest 🤷 ) with cds.enabled()
### Description

I have a piece of code that crashes unexpectedly during a pytest session and works fine in a separate IPython session,

```python
swgo_plot/plots/sky.py:447: in plot_sky_areas
    lon[k] = sky_coords[k].geocentrictrueecliptic.lon.deg
../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/coordinates/sky_coordinate.py:893: in __getattr__
    return self.transform_to(attr)
../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/coordinates/sky_coordinate.py:702: in transform_to
    new_coord = trans(self.frame, generic_frame)
../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/coordinates/transformations.py:1579: in __call__
    curr_coord = t(curr_coord, curr_toframe)
../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/coordinates/transformations.py:1145: in __call__
    return supcall(fromcoord, toframe)
../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/coordinates/builtin_frames/icrs_cirs_transforms.py:124: in icrs_to_gcrs
    gcrs_ra, gcrs_dec = atciqz(srepr.without_differentials(), astrom)
../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/coordinates/builtin_frames/utils.py:297: in atciqz
    pco = erfa.s2c(srepr.lon.radian, srepr.lat.radian)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Longitude 175. deg>, attr = 'radian'

    def __getattr__(self, attr):
        """
        Quantities are able to directly convert to other units that
        have the same physical type.
        """
        if not self._include_easy_conversion_members:
            raise AttributeError(
                f"'{self.__class__.__name__}' object has no '{attr}' member"
            )

        def get_virtual_unit_attribute():
            registry = get_current_unit_registry().registry
            to_unit = registry.get(attr, None)
            if to_unit is None:
                return None

            try:
                return self.unit.to(
                    to_unit, self.value, equivalencies=self.equivalencies
                )
            except UnitsError:
                return None

        value = get_virtual_unit_attribute()

        if value is None:
>           raise AttributeError(
                f"{self.__class__.__name__} instance has no attribute '{attr}'"
            )
E           AttributeError: Longitude instance has no attribute 'radian'

../../mambaforge/envs/swgo-plot/lib/python3.9/site-packages/astropy/units/quantity.py:1102: AttributeError
```

### Expected behavior

It should work the same in any environment if the package version is the same

### How to Reproduce

From a pdb session triggered by pytest I get that the triggering line is the following

```
-> lon[k] = sky_coords[k].geocentrictrueecliptic.lon.deg
(Pdb) k
0
(Pdb) sky_coords[k]
<SkyCoord (ICRS): (ra, dec) in deg
    (175., -0.5)>
```

indeed if (still inside pytest/pdb) I try to perform the transformation I get,

`(Pdb) sky_coords[k].geocentrictrueecliptic
*** AttributeError: Longitude instance has no attribute 'radian'`

To be sure I try to replicate the issue from scratch in a separate IPython session,

```ipython
In [1]: from astropy import units as u
   ...: from astropy.coordinates import SkyCoord
   ...: c = SkyCoord(ra=10.625*u.degree, dec=41.2*u.degree, frame='icrs')

In [2]: c.geocentrictrueecliptic
Out[2]:
<SkyCoord (GeocentricTrueEcliptic: equinox=J2000.000, obstime=J2000.000): (lon, lat, distance) in (deg, deg, )
    (27.76134106, 33.31188749, 1.)>

In [3]: c.geocentrictrueecliptic.lon.deg
Out[3]: 27.761341057229142
```

### Versions

macOS-11.7.4-x86_64-i386-64bit
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:01:00)
[Clang 13.0.1 ]
astropy 5.2.1
Numpy 1.23.4
pyerfa 2.0.0.1
Scipy 1.9.3
Matplotlib 3.6.2
pytest 7.2.0

### PR DESCRIPTION
**Pull request by HealthyPear**
### Description

When converting between coordinate frames what is currently happening is that `lon` and `lat` attributes get converted to `radians`, which depending on the current unit registry might or might not be defined as an alias to `rad`.

Luckily, the latter is always defined as the official name for radians, so it makes sense to make sure that we convert these quantities always using the guaranteed unit name.

Fixes #14580.

Note that #14933 was originally opened to update the documentation and recall the user to use a context manager when enabling e.g. CDS units (which do not define "radians", only "rad"). This PR should be the effective fix, but I think it still makes sense to have also the older PR.

Fixes #14580

### CODE REVIEW

**Review by eerovaher — CHANGES_REQUESTED** [review]
Overall this looks like a good bugfix, but it needs to be updated before it can be merged.

The `pre-commit.ci` test is failing. If you follow [our instructions how to enable `pre-commit` in your local repository](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit) then commits that would fail `pre-commit.ci` could be prevented from being created, and `pre-commit` can often automatically fix the problem that would cause `pre-commit.ci` to fail.

For bugfix pull requests I would prefer the test that reveals the bug to be in the first commit and the bugfix in the second. If done in that order it is simpler to verify that the bug indeed exists.

Some of the edited lines are not being executed during the tests. It would be good to have new tests to check those lines of code too.

Commit messages should be understandable on their own. Only adding a link to a issue on GitHub is not informative enough.

File: astropy/units/tests/test_units.py [hunk 1]
```diff
@@ -198,6 +198,14 @@ def test_cds_power():
     unit = u.Unit("10+22/cm2", format="cds", parse_strict="silent")
     assert unit.scale == 1e22
 
+def test_cds_with_coordinates_transformations():
```
[the remark this answers is not in the document]
Comment (HealthyPear) [reply astropy/units/tests/test_units.py:201]: in which module should I place it?

[the remark this answers is not in the document]
Comment (HealthyPear) [reply astropy/units/tests/test_units.py:201]: oh `test_transformations.py`

[the remark this answers is not in the document]
Comment (HealthyPear) [reply astropy/units/tests/test_units.py]: done

File: docs/changes/coordinates/14940.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Always convert explicitly longitudes and latitudes to 'rad' instead of 'radians' when transforming
+between coordinate frames. This fixes the cases in which other unit registries are enabled (such as CDS)
+for which 'radians' might not be defined.
```
[the remark this answers is not in the document]
Comment (HealthyPear) [reply docs/changes/coordinates/14940.bugfix.rst]: done

**Comment by HealthyPear** [comment]
Please, be aware that I have force-pushed the code-review commits in order to cleanup my branch's history.

File: astropy/coordinates/tests/test_transformations.py [hunk 3]
```diff
@@ -689,3 +689,13 @@ def test_combine_affine_params(first, second, check):
         assert result[1] is None
     else:
         assert_allclose(result[1].xyz, check[1].xyz)
+
+
+def test_cds_with_coordinates_transformations():
+    """See https://github.com/astropy/astropy/issues/14580."""
+    from astropy.coordinates import SkyCoord
```
Comment (eerovaher) [inline astropy/coordinates/tests/test_transformations.py:696]: The fact that `SkyCoord` is not being imported in this file suggests that perhaps `astropy/coordinates/tests/test_celestial_transformations.py` would be a more suitable location.

File: astropy/coordinates/tests/test_transformations.py [hunk 4]
```diff
@@ -689,3 +689,13 @@ def test_combine_affine_params(first, second, check):
...
+    from astropy.units import cds
```
Comment (eerovaher) [inline astropy/coordinates/tests/test_transformations.py:697]: As many imports as possible should be done in the beginning of the file.

**Review by eerovaher — CHANGES_REQUESTED** [review]
The first commit in a bugfix pull request should add a unit test that triggers the bug, the second commit should then fix it. The reason is that then it is very simple to verify that the new test actually triggers the bug, and that the fix actually addresses it. See also [relevant `astropy` developer documentation](https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#test-first-please).

The commit messages should also be improved. At the very least they shouldn't be factually wrong. For example, 661e91017ac1f0916c3a001382de06eadd5ee64d claims that a unit test is being moved under coordinates testing. That by itself is not very useful because that much could be seen just from the commit contents. It would be much more interesting to know why a test is being moved. But actually nothing is being moved because the unit test being added to `coordinates` testing is a brand new one. See also [the description of a good commit message in the Pro Git book](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project) (under the Commit Guidelines section).

**Pull request closed without merging**